### DISCUSSION TITLE
使用edgexfoundry / device-sdk-go。打包二进制运行报错

### DISCUSSION THREAD

**Discussion by 2105424002** [edgexfoundry/community#319] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/93776a57-2285-4975-8c96-a7aa866a05e5)
level=INFO ts=2024-12-03T06:52:47.245827685Z app=device-simple source=validation.go:35 msg="Responses to device validation requests will be published on topic: edgex/response/device-simple/<requestId>"
level=INFO ts=2024-12-03T06:52:47.246373025Z app=device-simple source=manager.go:128 msg="Metrics Manager started with a report interval of 30s"
level=INFO ts=2024-12-03T06:52:47.246418978Z app=device-simple source=clients.go:167 msg="Using REST for 'core-metadata' clients @ http://edgex-core-metadata:59881"
level=INFO ts=2024-12-03T06:52:47.24649528Z app=device-simple source=restrouter.go:56 msg="Registering v2 routes..."
level=ERROR ts=2024-12-03T06:52:47.598198564Z app=device-simple source=init.go:62 msg="Failed to init cache: request failed, status code: 502, err: "
level=ERROR ts=2024-12-03T06:52:47.598247118Z app=device-simple source=bootstrap.go:27 msg="Device Service bootstrapping failed"

**Comment by 2105424002** [comment]
我使用的版本是3.1.1。
用docker部署

**Comment by cloudxxx8** [comment]
EdgeX is a microservice architecture system, so it requires multiple services running together.
Device Service depends on core-metadata, and your Device Service can't reach core-metadata.
Please make sure other services are running and reachable.

**Reply by 2105424002** [reply]
我的所有服务运行正常。并且我已经设置了hosts

**Comment by 2105424002** [comment]
我按照https://docs.edgexfoundry.org/3.1/getting-started/Ch-GettingStartedSDK-Go/这个教程，重新运行了一个程序。现在报错信息为：
level=INFO ts=2024-12-03T08:14:43.974478497Z app=device-simple source=init.go:137 msg="Check core-metadata service's status by ping..."
level=INFO ts=2024-12-03T08:14:43.97616044Z app=device-simple source=init.go:56 msg="Service clients initialize successful."
level=INFO ts=2024-12-03T08:14:43.976227408Z app=device-simple source=restrouter.go:47 msg="Registering v2 routes..."
level=ERROR ts=2024-12-03T08:14:43.977796368Z app=device-simple source=init.go:42 msg="Failed to init cache: request failed, status code: 404, err: {\"message\":\"Not Found\"}\n"
level=INFO ts=2024-12-03T08:14:43.977884556Z app=device-simple source=messaging.go:89 msg="Disconnected from MessageBus"
level=INFO ts=2024-12-03T08:14:43.977939208Z app=device-simple source=httpserver.go:109 msg="Web server shutting down"
level=ERROR ts=2024-12-03T08:14:43.978051388Z app=device-simple source=httpserver.go:126 msg="Web server failed: http: Server closed"
level=INFO ts=2024-12-03T08:14:43.978087014Z app=device-simple source=httpserver.go:111 msg="Web server shut down"

**Comment by cloudxxx8** [comment]
please provide the full steps and environment setting you did

**Reply by 2105424002** [reply, marked as answer]
Thank you very much. This issue has been resolved. It's because the code pulled from the warehouse needs to have some things in the configuration file modified if it is packaged into a binary. When running in Docker, the key-value needs to be modified through the program on port 8500.
